Archive/Draft/RFCNNNN-PowerShell-LocalSession-Configuration.md

```diff
@@ -0,0 +1,192 @@
+
+### PowerShell `-ConfigurationFile` Command Line Switch
+
+The main proposal of this RFC is to add a new command line switch to `pwsh.exe` that takes an optional file path string to a PowerShell configuration (.pssc) file.
+This file would then be used to configure the default session runspace with the contained configuration information.
+The new command line switch would make configuring PowerShell running in server mode very easy.  
+
+For example, a PowerShell SSH connection endpoint is created by setting up a SSH 'subsystem' that runs `pwsh.exe`.
+Multiple PowerShell remoting endpoints could be created by using the new `-ConfigurationFile` command line switch:
+
+sshd_config file:
+
+```powershell
+...
+# override default of no subsystems
+#Subsystem  sftp        sftp-server
+# PowerShell endpoint configurations
+Subsystem   powershell1 /opt/microsoft/powershell/7/pwsh -S -NOLOGO -ConfigurationFile /etc/ssh/PSConfig1.pssc
+Subsystem   powershell2 /opt/microsoft/powershell/7/pwsh -S -NOLOGO -ConfigurationFile /etc/ssh/PSConfig2.pssc
+...
+```
+
+### Windows/WinRM Configuration Elements
+
+Unfortunately, some of the configuration elements supported in a .pssc configuration file is for Windows/WinRM remote connections only.
+For example, `virtual` and `group managed` accounts are Windows platform only.
+And JEA role definitions are based on Windows platform accounts and groups.  
+
+We want to separate Windows/WinRM specific configuration elements from what PowerShell supports implicitly.
+The new `-ConfigurationFile` command line switch and policy setting will support PowerShell configurations only, and will throw a terminating error if unsupported WinRM/Windows configurations are specified.  
+
+The following are Windows/WinRM configuration elements that will **not** be supported for local sessions:
+
+```powershell
+RunAsVirtualAccount
+RunAsVirtualAccountGroups
+GroupManagedServiceAccount
+RoleDefinitions
+RequiredGroups
+PowerShellVersion
+```
+
+### PowerShell Configuration Elements
+
+The following are configuration elements that **will** be supported:
+
+```powershell
+SessionType
+TranscriptDirectory
+MountUserDrive
+UserDriveMaximumSize
+ScriptsToProcess
+LanguageMode
+ExecutionPolicy
+ModulesToImport
+VisibleAliases
+VisibleCmdlets
+VisibleFunctions
+VisibleExternalCommands
+VisibleProviders
+AliasDefinitions
+FunctionDefinitions
+VariableDefinitions
+EnvironmentVariables
+TypesToProcess
+FormatsToProcess
+AssembliesToLoad
+```
+
+## Alternate Design Options
+
+### PowerShell `SessionConfiguration` Policy Setting
+
+Additional to the `-ConfigurationFile` command line switch, session configuration could also be settable via Windows Group Policy or PowerShell configuration file.
+The policy session configuration would be specified by either a configuration file path or a string defining the configuration.
```

Comment:
I think Windows Group Policy is useful.

Reply:
I agree.  But it was decided that this is not needed initially, and could be added later.

Reply:
Perhaps mention that after `Configuration Through Application Control Policies`??